static/js/settings_org.js

```diff
@@ -841,16 +861,9 @@ export function check_property_changed(elem, for_realm_default_settings, sub) {
             proposed_val = JSON.stringify(proposed_val);
             break;
         case "realm_notifications_stream_id":
-            proposed_val = Number.parseInt(notifications_stream_widget.value(), 10);
-            break;
         case "realm_signup_notifications_stream_id":
-            proposed_val = Number.parseInt(signup_notifications_stream_widget.value(), 10);
-            break;
         case "realm_default_code_block_language":
-            proposed_val = default_code_language_widget.value();
-            if (proposed_val.length === 0) {
-                proposed_val = null;
-            }
+            proposed_val = get_dropdown_list_widget_setting_value($elem, false);
```

Comment:
Hmm, could this part be changed from a list of specific fields to something keyed off the widget type? 

It'd be nice if we didn't need to anything special inside this file to add a new one.

Reply:
Along with `for_api_data`, the other thing to consider is that we pass `typeof current_val` when calling `get_input_element_value` in the `default` case and it results in some bugs if we directly replace it with widget-type. I was thinking of just always using the widget-type always instead, will need some careful refactoring but should not too difficult. Will start work on this soon.